python/shotgun_desktop/startup.py

```diff
@@ -775,6 +814,14 @@ def main(**kwargs):
             logger.info("Login canceled. Quitting.")
             return 0
 
+        # In the case where the site is using SSO, the user needs to renew
+        # its claims regularily. So we kick off a separate newewal thread.
+        if user._impl.get_saml_expiration() is not None:
```

Comment:
I mean, technically, we are guaranteed that we know which version of the API is bundled for authentication, so we can in theory violate the API, but that's not a good practice. Let's discuss this.